src/box/applier.cc

```diff
@@ -1434,8 +1436,8 @@ applier_disconnect(struct applier *applier, enum applier_state state)
 		fiber_join(applier->writer);
 		applier->writer = NULL;
 	}
-
-	coio_close_io(loop(), &applier->io);
+	if (iostream_is_initialized(&applier->io))
+		iostream_close(&applier->io);
```

Comment:
There's an assertion - it's in `iostream_destroy`, which is called by `iostream_close`.

I don't think it's a good idea to make `iostream_close` (and/or `iostream_destroy`) nop in case the stream is not initialized, because this would make the code using streams less clear re stream lifetime IMO.